### PR TITLE
 [FLINK-13273][sql-client] Allow switching planners in SQL Client

### PR DESCRIPTION
**Pull request by twalthr**
## What is the purpose of the change

This allows to switch planners in SQL Client. A user can specify `execution.planner=blink` to activate it. The SQL Client E2E tests as well as integration tests have been updated for verifying the behavior.

This PR is based on #9263.

## Brief change log

See commit messages.

## Verifying this change

This change is already covered by existing tests, such as `LocalExecutorITCase`.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): yes
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? yes
  - If yes, how is the feature documented? docs

### CODE REVIEW

**Comment by dawidwys** [comment]
+1 for this PR. I think though we need to change the #9263 first a bit.

File: flink-end-to-end-tests/run-nightly-tests.sh [hunk 1]
```diff
@@ -143,7 +143,8 @@ run_test "Avro Confluent Schema Registry nightly end-to-end test" "$END_TO_END_D
 run_test "State TTL Heap backend end-to-end test" "$END_TO_END_DIR/test-scripts/test_stream_state_ttl.sh file"
 run_test "State TTL RocksDb backend end-to-end test" "$END_TO_END_DIR/test-scripts/test_stream_state_ttl.sh rocks"
 
-run_test "SQL Client end-to-end test" "$END_TO_END_DIR/test-scripts/test_sql_client.sh"
+run_test "SQL Client end-to-end test (Old planner)" "$END_TO_END_DIR/test-scripts/test_sql_client.sh old"
```
Comment (docete) [inline flink-end-to-end-tests/run-nightly-tests.sh]: Should add blink planner SQL Client e2e test in tools/travis/splits/split_misc.sh

Comment (twalthr) [reply flink-end-to-end-tests/run-nightly-tests.sh]: Thanks for the hint. Yes makes sense.

**Comment by twalthr** [comment]
Thanks for the feedback @dawidwys and @docete. I updated the PR. It was also possible to make the SQL Client work without changes to the factories. Please take another look.

**Comment by dawidwys** [comment]
@twalthr Thank you for solving the factories PR. I think now this PR is good to be merged.

**Pull request closed without merging**